### PR TITLE
ne 3.0.1 (new formula)

### PR DESCRIPTION
**Pull request by alehed**
This adds a formula for the ne text editor. Ne is a lightweight
text editor with a focus on performance and little bandwidth use.
Apart from that it is designed to be approachable yet extensible
enough to keep an advanced user productive.
Development is located at https://github.com/vigna/ne.

### CODE REVIEW

File: Library/Formula/ne.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Ne < Formula
+  desc "ne, the nice editor"
+  homepage "http://ne.di.unimi.it"
+  url "http://ne.di.unimi.it/ne-3.0.1.tar.gz"
+  sha256 "92b646dd2ba64052e62deaa4239373821050a03e1b7d09d203ce04f2adfbd0e4"
+
+  bottle :unneeded
```
Comment (dunn) [inline Library/Formula/ne.rb]: Why shouldn't it be bottled?

File: Library/Formula/ne.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  def install
+    system "make"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "#{bin}/ne", "--help"
```
Comment (dunn) [inline Library/Formula/ne.rb]: We'd love a test that checks a bit more of the functionality!

File: Library/Formula/ne.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Ne < Formula
+  desc "ne, the nice editor"
```
Comment (DomT4) [inline Library/Formula/ne.rb]: Please don't start formula descriptions with the formula name. That happens automatically.

**Comment by alehed** [comment]
This is my first formula so bear with me.
Found a way to test the core editing functionality of the editor. I picked up the bottle :unneeded from some other formula and I didn't see any use in providing a precompiled keg if it's not hosted anywhere.

**Comment by alehed** [comment]
It seems like the files could not be created on travis. I'll have a look at it afterwards

**Comment by alehed** [comment]
For now I can't determine what the problem is exactly. The block is caused by the editor asking for user feedback which happens every time it can't write to the file or there file was written by another application in the meantime.

**Comment by DomT4** [comment]
Please be careful how many pushes you make here; each push is ran through our CI and doing so many is quite resource intensive.

You can try adding `ENV["TERM"] = "xterm"` to the top of `test do`, which has helped in similar errors before.

**Comment by alehed** [comment]
Ok, I'll give this one more try and then I forfeit.

**Comment by DomT4** [comment]
I can't get it to fail locally, so I do suspect `ENV["TERM"] = "xterm"` will help.

**Comment by alehed** [comment]
Unfortunately ne has no command line options that avoid opening the editor except --help. The only thing that seems possible to me is different file permissions on travis, because it works just fine on my machine.

**Comment by hussain0ip** [comment]
hi guys 
i need some help for installing android stuido

**Comment by hussain0ip** [comment]
please speek with me on tiwtter @thesnipergreat

**Comment by alehed** [comment]
Looks like you're in the wrong thread

File: Library/Formula/ne.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  def install
+    system "make"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    ENV["TERM"] = "xterm"
+    (testpath/"test.txt").write("This is a test document.\n")
+    (testpath/"test_ne").write("GotoLine 2\nInsertString line 2\nExit\n")
+    system "#{bin}/ne", "--macro", "#{(testpath/"test_ne")}", "#{(testpath/"test.txt")}"
```
Comment (dunn) [inline Library/Formula/ne.rb]: What may work is changing this line to:
```ruby
system "script", "-q", "/dev/null", "#{bin}/ne", "--macro", "#{(testpath/"test_ne")}", "#{(testpath/"test.txt")}"
```
to trick `ne` into thinking input is coming from a tty.

**Comment by alehed** [comment]
The CI part passed! Thanks a lot @dunn @DomT4 for the support, couldn't have done it on my own!

File: Library/Formula/ne.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class Ne < Formula
+  desc "The nice editor"
+  homepage "http://ne.di.unimi.it"
+  url "http://ne.di.unimi.it/ne-3.0.1.tar.gz"
+  sha256 "92b646dd2ba64052e62deaa4239373821050a03e1b7d09d203ce04f2adfbd0e4"
+
```
Comment (dunn) [inline Library/Formula/ne.rb]: Couldn't hurt to add:
```ruby
head "https://github.com/vigna/ne.git"
```
as well

Comment (dunn) [inline Library/Formula/ne.rb]: Oh, building from the repository requires LaTeX for the documentation. Nevermind, then.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**